knack/cli.py

```diff
@@ -91,8 +93,28 @@ def __init__(self,
         self.output = self.output_cls(cli_ctx=self)
         self.result = None
         self.query = query_cls(cli_ctx=self)
+
+        # As logging is initialized in `invoke`, call `logger.debug` or `logger.info` here won't work.
+        self.init_debug_log = []
+        self.init_info_log = []
```

Comment:
Add lists for debug and info logs which can't be printed with `logger.debug` and `logger.info` yet, until the logger has been initialized.